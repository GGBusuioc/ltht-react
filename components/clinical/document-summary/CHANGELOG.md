# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.0.4](https://github.com/ltht-epr/ltht-react/compare/@ltht-react/document-summary@0.0.3...@ltht-react/document-summary@0.0.4) (2020-04-08)

**Note:** Version bump only for package @ltht-react/document-summary






## [0.0.3](https://ssh.github.com/ltht-epr/ltht-react/compare/@ltht-react/document-summary@0.0.2...@ltht-react/document-summary@0.0.3) (2020-04-06)

**Note:** Version bump only for package @ltht-react/document-summary





## [0.0.2](https://ssh.github.com/ltht-epr/ltht-react/compare/@ltht-react/document-summary@0.0.1...@ltht-react/document-summary@0.0.2) (2020-04-03)

**Note:** Version bump only for package @ltht-react/document-summary





## 0.0.1 (2020-04-03)

**Note:** Version bump only for package @ltht-react/document-summary
