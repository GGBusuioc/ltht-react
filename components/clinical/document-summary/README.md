# DocumentSummary

<!-- STORY -->

### Import

```js
import DocumentSummary from '@ltht-react/document-summary'
```

### Usage

```jsx
<DocumentSummary documents={documentReferences} />
```

### Properties

| Prop        | Required | Default | Type                | Description                     |
| :---------- | :------- | :------ | :------------------ | :------------------------------ |
| `documents` | No       |         | DocumentReference[] | An array of document references |
