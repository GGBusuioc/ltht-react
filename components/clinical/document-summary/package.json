{
  "name": "@ltht-react/document-summary",
  "version": "0.0.17",
  "description": "ltht-react clinical DocumentSummary component.",
  "author": "LTHT",
  "homepage": "",
  "license": "MIT",
  "main": "lib/index.js",
  "typings": "lib/index.d.ts",
  "files": [
    "lib"
  ],
  "directories": {
    "lib": "lib"
  },
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "clean": "rimraf lib",
    "build": "yarn build:lib",
    "watch": "tsc -w",
    "build:lib": "tsc",
    "type:check": "tsc --noEmit",
    "lint": "prettylint 'src/*.{ts,tsx}'"
  },
  "dependencies": {
    "@emotion/core": "^10.0.27",
    "@emotion/styled": "^10.0.27",
    "@ltht-react/icon": "^0.1.15",
    "@ltht-react/list": "^0.1.13",
    "@ltht-react/styles": "^0.1.7",
    "@ltht-react/summary": "^0.0.11",
    "@ltht-react/types": "^0.0.14",
    "@ltht-react/utils": "^0.1.7",
    "emotion": "^10.0.27",
    "react": "^16.12.0"
  },
  "gitHead": "b259430b592502d8447f782482b2c8fd74e78fed"
}
