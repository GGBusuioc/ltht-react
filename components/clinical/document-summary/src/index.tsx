/** @jsx jsx */
import React from 'react'
import { css, jsx } from '@emotion/core'

import { DocumentReference } from '@ltht-react/types'
import { CARD_BORDER_COLOUR, TEXT_COLOURS } from '@ltht-react/styles'
import { UnorderedList, ListItem } from '@ltht-react/list'
import DocumentSummaryItem from './document-summary-item'
import DocumentSummaryContent from './document-summary-content'

const styles = {
  root: css`
    margin-right: -0.5rem;
    > ul li:first-of-type {
      font-weight: bold;
    }
    > ul li:not(:first-of-type) {
      border-top: 1px solid ${CARD_BORDER_COLOUR};
      padding-top: 0.25rem;
    }
    > ul li {
      align-items: center;
      display: flex;
      flex-direction: row;
      flex-wrap: nowrap;
      padding-bottom: 0.25rem;
      padding-right: 0.5rem;
      > div {
        flex: auto;
        padding-left: 1rem;
      }
      > div:first-of-type {
        flex: 0 0 6rem;
        padding-left: 0;
      }
      > div.source {
        flex: 0 0 15rem;
        > div:last-of-type {
          color: ${TEXT_COLOURS.SECONDARY.VALUE};
          font-weight: normal;
        }
      }
      > div:last-of-type {
        flex: 0 0 2.25rem;
      }
    }
  `,
}

const DocumentSummary: React.FC<Props> = ({ documents = undefined }) => {
  if (!documents) {
    return null
  }

  const items = documents.map(x => <DocumentSummaryItem document={x} />)

  return (
    <div css={styles.root}>
      <UnorderedList>
        <ListItem key="0">
          <DocumentSummaryContent
            created="Created"
            description="Description"
            source="Source"
            status="Status"
            clickable={false}
          />
        </ListItem>
        {items}
      </UnorderedList>
    </div>
  )
}

interface Props {
  documents: DocumentReference[] | undefined
}

export default DocumentSummary
