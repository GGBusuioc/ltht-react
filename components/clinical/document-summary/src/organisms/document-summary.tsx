import { FC, HTMLAttributes } from 'react'
import styled from '@emotion/styled'

import { DocumentReference } from '@ltht-react/types'

import Date from '../atoms/document-date'
import Description from '../atoms/document-description'
import Source from '../atoms/document-source'
import Status from '../atoms/document-status'
import Redacted from '../molecules/document-redacted'

const StyledSummary = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
`

const StyledDate = styled.div``

const StyledDescription = styled.div`
  flex-grow: 1;
  margin-left: 0.5rem;
`

const StyledSource = styled.div`
  display: flex;
  flex-direction: column;
  text-align: right;
`

const DocumentSummary: FC<Props> = ({ document, ...rest }) => {
  if (document.metadata.isRedacted) {
    return (
      <StyledSummary {...rest}>
        <Redacted document={document} />
      </StyledSummary>
    )
  }

  return (
    <StyledSummary {...rest}>
      <StyledDate>
        <Date document={document} />
      </StyledDate>
      <StyledDescription>
        <Description document={document} />
      </StyledDescription>
      <StyledSource>
        <Source document={document} />
        <Status document={document} />
      </StyledSource>
    </StyledSummary>
  )
}

interface Props extends HTMLAttributes<HTMLDivElement> {
  document: DocumentReference
}

export default DocumentSummary
