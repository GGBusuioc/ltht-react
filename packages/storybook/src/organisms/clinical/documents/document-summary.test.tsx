import React from 'react'
import { mount } from 'enzyme'
import ReactDOM from 'react-dom'

import DocumentSummary from '@ltht-react/document-summary'
import { Document1, Document2, Document3 } from './document-summary.fixture'

describe('DocumentSummaryy', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<DocumentSummary documents={[Document1, Document2, Document3]} />, div)
  })
  it('matches snapshot', () => {
    expect(mount(<DocumentSummary documents={[Document1, Document2, Document3]} />)).toMatchSnapshot('wrapper mount')
  })
})
