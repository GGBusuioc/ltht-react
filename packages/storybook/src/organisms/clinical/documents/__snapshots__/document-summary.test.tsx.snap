// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentSummaryy matches snapshot: wrapper mount 1`] = `
.emotion-7 {
  margin-right: -0.5rem;
}

.emotion-7 > ul li:first-of-type {
  font-weight: bold;
}

.emotion-7 > ul li:not(:first-of-type) {
  border-top: 1px solid #b0b0b0;
  padding-top: 0.25rem;
}

.emotion-7 > ul li {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  padding-bottom: 0.25rem;
  padding-right: 0.5rem;
}

.emotion-7 > ul li > div {
  -webkit-flex: auto;
  -ms-flex: auto;
  flex: auto;
  padding-left: 1rem;
}

.emotion-7 > ul li > div:first-of-type {
  -webkit-flex: 0 0 6rem;
  -ms-flex: 0 0 6rem;
  flex: 0 0 6rem;
  padding-left: 0;
}

.emotion-7 > ul li > div.source {
  -webkit-flex: 0 0 15rem;
  -ms-flex: 0 0 15rem;
  flex: 0 0 15rem;
}

.emotion-7 > ul li > div.source > div:last-of-type {
  color: #768692;
  font-weight: normal;
}

.emotion-7 > ul li > div:last-of-type {
  -webkit-flex: 0 0 2.25rem;
  -ms-flex: 0 0 2.25rem;
  flex: 0 0 2.25rem;
}

.emotion-6 {
  list-style-type: none;
}

.emotion-0 {
  color: -v-a-l-u-e:#768692;
  -d-a-r-k-e-r25: #59656e;
  -l-i-g-h-t-e-r25: #98a4ad;
}

<DocumentSummary
  documents={
    Array [
      Object {
        "content": Array [],
        "created": Object {
          "value": "2020-04-01T00:00:00+00:00",
        },
        "description": "Description of document one",
        "id": "6b554095-bd5f-4134-8dbb-a21971959fd2",
        "indexed": Object {
          "value": "2020-04-01T00:00:00+00:00",
        },
        "metadata": Object {
          "dataSources": Array [
            Object {
              "display": "Mock",
            },
          ],
          "isRedacted": false,
          "requestedWhen": "",
        },
        "status": "CURRENT",
        "type": Object {},
      },
      Object {
        "content": Array [],
        "created": Object {
          "value": "2019-10-11T00:00:00+00:00",
        },
        "description": "Description of document two",
        "id": "2d889a55-d7ec-46ae-b674-23dfc6666018",
        "indexed": Object {
          "value": "2019-10-11T00:00:00+00:00",
        },
        "metadata": Object {
          "dataSources": Array [
            Object {
              "display": "Mock",
            },
          ],
          "isRedacted": false,
          "requestedWhen": "",
        },
        "status": "ENTERED_IN_ERROR",
        "type": Object {},
      },
      Object {
        "content": Array [],
        "created": Object {
          "value": "2020-01-23T00:00:00+00:00",
        },
        "description": "Description of document three",
        "id": "38962be0-4e5e-4640-a23e-af45da7c0578",
        "indexed": Object {
          "value": "2020-01-23T00:00:00+00:00",
        },
        "metadata": Object {
          "dataSources": Array [
            Object {
              "display": "Mock",
            },
          ],
          "isRedacted": false,
          "requestedWhen": "",
        },
        "status": "SUPERSEDED",
        "type": Object {},
      },
    ]
  }
>
  <div
    className="emotion-7"
  >
    <UnorderedList>
      <ul
        className="emotion-6"
      >
        <ListItem
          key="0"
        >
          <li>
            <DocumentSummaryContent
              clickable={false}
              created="Created"
              description="Description"
              source="Source"
              status="Status"
            >
              <div>
                Created
              </div>
              <div>
                Description
              </div>
              <div
                className="source"
              >
                <div>
                  Source
                </div>
                <div>
                  Status
                </div>
              </div>
              <div />
            </DocumentSummaryContent>
          </li>
        </ListItem>
        <DocumentSummaryItem
          document={
            Object {
              "content": Array [],
              "created": Object {
                "value": "2020-04-01T00:00:00+00:00",
              },
              "description": "Description of document one",
              "id": "6b554095-bd5f-4134-8dbb-a21971959fd2",
              "indexed": Object {
                "value": "2020-04-01T00:00:00+00:00",
              },
              "metadata": Object {
                "dataSources": Array [
                  Object {
                    "display": "Mock",
                  },
                ],
                "isRedacted": false,
                "requestedWhen": "",
              },
              "status": "CURRENT",
              "type": Object {},
            }
          }
        >
          <ListItem
            key="6b554095-bd5f-4134-8dbb-a21971959fd2"
          >
            <li>
              <DocumentSummaryContent
                clickable={true}
                created="01/04/2020"
                description="Description of document one"
                source="[SOURCE HERE]"
                status="CURRENT"
              >
                <div>
                  01/04/2020
                </div>
                <div>
                  Description of document one
                </div>
                <div
                  className="source"
                >
                  <div>
                    [SOURCE HERE]
                  </div>
                  <div>
                    CURRENT
                  </div>
                </div>
                <div>
                  <ChevronIcon
                    direction="right"
                    size="large"
                  >
                    <FontAwesomeIcon
                      border={false}
                      className="emotion-0"
                      fixedWidth={false}
                      flip={null}
                      icon={
                        Object {
                          "icon": Array [
                            320,
                            512,
                            Array [],
                            "f054",
                            "M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z",
                          ],
                          "iconName": "chevron-right",
                          "prefix": "fas",
                        }
                      }
                      inverse={false}
                      listItem={false}
                      mask={null}
                      pull={null}
                      pulse={false}
                      rotation={null}
                      size="lg"
                      spin={false}
                      swapOpacity={false}
                      symbol={false}
                      title=""
                      transform={null}
                    >
                      <svg
                        aria-hidden="true"
                        className="svg-inline--fa fa-chevron-right fa-w-10 fa-lg emotion-0"
                        data-icon="chevron-right"
                        data-prefix="fas"
                        focusable="false"
                        role="img"
                        style={Object {}}
                        viewBox="0 0 320 512"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
                          fill="currentColor"
                          style={Object {}}
                        />
                      </svg>
                    </FontAwesomeIcon>
                  </ChevronIcon>
                </div>
              </DocumentSummaryContent>
            </li>
          </ListItem>
        </DocumentSummaryItem>
        <DocumentSummaryItem
          document={
            Object {
              "content": Array [],
              "created": Object {
                "value": "2019-10-11T00:00:00+00:00",
              },
              "description": "Description of document two",
              "id": "2d889a55-d7ec-46ae-b674-23dfc6666018",
              "indexed": Object {
                "value": "2019-10-11T00:00:00+00:00",
              },
              "metadata": Object {
                "dataSources": Array [
                  Object {
                    "display": "Mock",
                  },
                ],
                "isRedacted": false,
                "requestedWhen": "",
              },
              "status": "ENTERED_IN_ERROR",
              "type": Object {},
            }
          }
        >
          <ListItem
            key="2d889a55-d7ec-46ae-b674-23dfc6666018"
          >
            <li>
              <DocumentSummaryContent
                clickable={true}
                created="11/10/2019"
                description="Description of document two"
                source="[SOURCE HERE]"
                status="ENTERED_IN_ERROR"
              >
                <div>
                  11/10/2019
                </div>
                <div>
                  Description of document two
                </div>
                <div
                  className="source"
                >
                  <div>
                    [SOURCE HERE]
                  </div>
                  <div>
                    ENTERED_IN_ERROR
                  </div>
                </div>
                <div>
                  <ChevronIcon
                    direction="right"
                    size="large"
                  >
                    <FontAwesomeIcon
                      border={false}
                      className="emotion-0"
                      fixedWidth={false}
                      flip={null}
                      icon={
                        Object {
                          "icon": Array [
                            320,
                            512,
                            Array [],
                            "f054",
                            "M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z",
                          ],
                          "iconName": "chevron-right",
                          "prefix": "fas",
                        }
                      }
                      inverse={false}
                      listItem={false}
                      mask={null}
                      pull={null}
                      pulse={false}
                      rotation={null}
                      size="lg"
                      spin={false}
                      swapOpacity={false}
                      symbol={false}
                      title=""
                      transform={null}
                    >
                      <svg
                        aria-hidden="true"
                        className="svg-inline--fa fa-chevron-right fa-w-10 fa-lg emotion-0"
                        data-icon="chevron-right"
                        data-prefix="fas"
                        focusable="false"
                        role="img"
                        style={Object {}}
                        viewBox="0 0 320 512"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
                          fill="currentColor"
                          style={Object {}}
                        />
                      </svg>
                    </FontAwesomeIcon>
                  </ChevronIcon>
                </div>
              </DocumentSummaryContent>
            </li>
          </ListItem>
        </DocumentSummaryItem>
        <DocumentSummaryItem
          document={
            Object {
              "content": Array [],
              "created": Object {
                "value": "2020-01-23T00:00:00+00:00",
              },
              "description": "Description of document three",
              "id": "38962be0-4e5e-4640-a23e-af45da7c0578",
              "indexed": Object {
                "value": "2020-01-23T00:00:00+00:00",
              },
              "metadata": Object {
                "dataSources": Array [
                  Object {
                    "display": "Mock",
                  },
                ],
                "isRedacted": false,
                "requestedWhen": "",
              },
              "status": "SUPERSEDED",
              "type": Object {},
            }
          }
        >
          <ListItem
            key="38962be0-4e5e-4640-a23e-af45da7c0578"
          >
            <li>
              <DocumentSummaryContent
                clickable={true}
                created="23/01/2020"
                description="Description of document three"
                source="[SOURCE HERE]"
                status="SUPERSEDED"
              >
                <div>
                  23/01/2020
                </div>
                <div>
                  Description of document three
                </div>
                <div
                  className="source"
                >
                  <div>
                    [SOURCE HERE]
                  </div>
                  <div>
                    SUPERSEDED
                  </div>
                </div>
                <div>
                  <ChevronIcon
                    direction="right"
                    size="large"
                  >
                    <FontAwesomeIcon
                      border={false}
                      className="emotion-0"
                      fixedWidth={false}
                      flip={null}
                      icon={
                        Object {
                          "icon": Array [
                            320,
                            512,
                            Array [],
                            "f054",
                            "M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z",
                          ],
                          "iconName": "chevron-right",
                          "prefix": "fas",
                        }
                      }
                      inverse={false}
                      listItem={false}
                      mask={null}
                      pull={null}
                      pulse={false}
                      rotation={null}
                      size="lg"
                      spin={false}
                      swapOpacity={false}
                      symbol={false}
                      title=""
                      transform={null}
                    >
                      <svg
                        aria-hidden="true"
                        className="svg-inline--fa fa-chevron-right fa-w-10 fa-lg emotion-0"
                        data-icon="chevron-right"
                        data-prefix="fas"
                        focusable="false"
                        role="img"
                        style={Object {}}
                        viewBox="0 0 320 512"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
                          fill="currentColor"
                          style={Object {}}
                        />
                      </svg>
                    </FontAwesomeIcon>
                  </ChevronIcon>
                </div>
              </DocumentSummaryContent>
            </li>
          </ListItem>
        </DocumentSummaryItem>
      </ul>
    </UnorderedList>
  </div>
</DocumentSummary>
`;
